redis_api/api/routes.py

```diff
@@ -43,20 +46,18 @@ def get_candidates(candidate_id, serve_fake=False):
     :return: data on all candidates or for one specific candidate
     :rtype: JSON
     """
-    if serve_fake:
-        return fake_data.get_candidates_shape()
     try:
-        response = r.execute_command("JSON.GET", "Candidates")
-        if not response:
-            return empty_response("Candidates")
-        cand_json = json.loads(response)
+        redis = RedisClient()
+        candidateShape = redis.getCandidateShape()
         if candidate_id is None:
-            return jsonify({"Candidates": cand_json})
+            # TODO: Sync up with Front-End for the spec
+            if candidateShape:
+                return jsonify(candidateShape)
         else:
-            for candidate in cand_json:
-                if candidate["ID"] == candidate_id:
-                    return candidate
-            return empty_response(candidate_id)
+            if candidateShape:
```

Comment:
If this statement is False, nothing will return from the function. I suggest checking candidateShape first, then nesting the other conditionals inside of it.

Reply:
Done

Reply:
Done